### PR TITLE
render now as access to the res object

### LINKED ISSUE #9: Pass the req object when rendering a template
This way functions have a lot more flexibility

### PR DESCRIPTION
**Pull request by earobinson**
- This allows render to use functions to achieve more complex results
- Cleaned up the way that models are used in tests
- Updated jyson and mocha to the latest
- Updated travis to report to coveralls
  - Added a coveralls badge to the repo
- Download badge now links to npm-stats

Closes #9

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -14,4 +14,5 @@ before_install:
   - npm i mongodb-download@2.2.3
 script:
   - npm run test:lint
-  - npm run test
\ No newline at end of file
+  - npm run test:coverage
+  - npm run report:coverage
```
Comment (misshula) [inline .travis.yml]: just an easy way to report cov? or more to it?

**Review by misshula — APPROVED** [review]
this is great.

File: .travis.yml (hunk 1 above)
Comment (earobinson) [reply .travis.yml]: its the same pattern as jyson, just gives you npm scrips that ci can run to report coverage

**Comment by earobinson** [comment]
Going merge this as it has one ✅  would still love a review @azharkhan

**Pull request merged**